crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +88,131 @@ pub struct Search {
-
-                    if let Some(desc) = entry.get("description").unwrap().as_str() {
-                        print!(" - {}", desc)
-                    }
-                    println!();
-                }
-            }
+            // FIXME: We may have warnings on `stderr` even with a successful call to `pkgdb`.
+            //        We aren't checking that at all at the moment because better overall error handling
+            //        is coming in a later PR.
+            let search_results = SearchResults::try_from(output.stdout.as_slice())?;
+            render_search_results(search_results, self.json)?;
             Ok(())
         } else {
+            let err_msg = String::from_utf8_lossy(&output.stdout);
             bail!(
-                "pkgdb exited with status code {}",
-                output.status.code().unwrap_or(-1)
+                "pkgdb exited with status code {}: {}",
+                output.status.code().unwrap_or(-1),
+                err_msg
             );
         }
     }
 }
 
+fn construct_search_params(search_term: &Option<String>, flox: &Flox) -> Result<SearchParams> {
+    let channels = flox
+        .channels
+        .iter()
+        .filter_map(|entry| {
+            if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
+                None
+            } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
+                Some((ChannelType::Flox, entry))
+            } else {
+                Some((ChannelType::User, entry))
+            }
+        })
+        .sorted();
+
+    // Create `registry` parameter for `pkgdb`
+    let mut priority: Vec<String> = Vec::new();
+    let mut inputs = HashMap::new();
+    for (_, entry) in channels {
+        priority.push(entry.from.id.to_string());
+        let input = RegistryInput {
+            from: FlakeRef::Indirect(entry.from.clone()),
+            // TODO: handle `subtrees` and `stabilities`
+            subtrees: None,
+            stabilities: None,
+        };
+        inputs.insert(entry.from.id.to_string(), input);
+    }
+    let registry = Registry {
+        inputs,
+        priority,
+        defaults: RegistryDefaults::default(),
+    };
+
+    // We've already checked that the search term is Some(_)
+    let query = Query::from_str(search_term.as_ref().unwrap())?;
+
+    Ok(SearchParams {
+        registry,
+        query,
+        ..SearchParams::default()
+    })
+}
+
+// This is likely to change significantly after the output format of search results is specced out
+fn render_search_results(search_results: SearchResults, as_json: bool) -> Result<()> {
+    if as_json {
+        let json = serde_json::to_string(&search_results.results)?;
+        println!("{}", json);
+        return Ok(());
+    }
+    let summarized_results = search_results
+        .results
+        .iter()
+        .map(|r| {
+            let path_components = r.attr_path.clone();
+            let flake_attr = if !HIDDEN_CHANNELS.contains_key(r.input.as_str()) {
```

Comment:
Can you explain what's going on with `HIDDEN_CHANNELS`?

Reply:
See my response to the comment about `ChannelType`.